app/lib/gateway_settings/proxy_extension.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module GatewaySettings
+  module ProxyExtension
+    extend ActiveSupport::Concern
+    included do
+      has_one :gateway_configuration, dependent: :delete, inverse_of: :proxy, autosave: true
```

Comment:
why `dependent: :delete` instead of `dependent: :destroy` ? :thinking:

Reply:
Because there is nothing else we can do, it is just a storage, no other dependencies nor callbacks

Basically I want to extract all setttings (only data) from the Proxy model
Proxy should have mandatory data, the settings are just configurations